src/imagery/i.eodag/i.eodag.py

```diff
@@ -825,33 +822,32 @@ def sort_result(search_result):
     :return: Sorted EO products
     :rtype: class:'eodag.api.search_result.SearchResult'
     """
-    sort_keys = options["sort"].split(",")
-    sort_order = options["order"]
-
-    # Sort keys and sort orders are matched respectively
-    def products_compare(first, second):
-        for sort_key in sort_keys:
-            if sort_key == "ingestiondate":
-                if "startTimeFromAscendingNode" not in first.properties:
-                    continue
-                first_value = first.properties["startTimeFromAscendingNode"]
-                second_value = second.properties["startTimeFromAscendingNode"]
-            elif sort_key == "cloudcover":
-                if "cloudCover" not in first.properties:
-                    continue
-                first_value = first.properties["cloudCover"]
-                second_value = second.properties["cloudCover"]
-            elif sort_key == "footprint":
-                # Sort by title lexicographically
-                first_value = first.properties["title"]
-                second_value = second.properties["title"]
-            if first_value < second_value:
-                return 1 if sort_order == "desc" else -1
-            elif first_value > second_value:
-                return -1 if sort_order == "desc" else 1
-        return 0
-
-    search_result.sort(key=cmp_to_key(products_compare))
+    sort_keys = list()
+    for sort_key in options["sort"].split(","):
+        if sort_key == "ingestiondate":
+            sort_keys.append(
+                (
+                    "startTimeFromAscendingNode",
+                    (
+                        "3000-12-31T00:00:00"
```

Comment:
@HamedElgizery maybe we could use datetime.today() + datetime.timdelta(1) to non use a fixed data

Reply:
Another approach without artificial keys could be like this: https://stackoverflow.com/a/18411610